packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/migrate-idp-resources.ts

```diff
@@ -0,0 +1,61 @@
+import { $TSObject, JSONUtilities, pathManager, Template } from '@aws-amplify/amplify-cli-core';
+import { CognitoIdentityServiceProvider } from 'aws-sdk';
+import { getUserPoolId } from './get-user-pool-id';
+
+const { readJson } = JSONUtilities;
+const { getCurrentCfnTemplatePathFromBuild } = pathManager;
+
+export const migrateResourcesToCfn = (resourceName: string): boolean => {
+  const authCfnTemplatePath = getCurrentCfnTemplatePathFromBuild('auth', resourceName);
+  const authCfnTemplate: Template | undefined = readJson(authCfnTemplatePath, { throwIfNotExist: false });
+  const lambdaCalloutCreatedInCloud = authCfnTemplate?.Resources?.HostedUIProvidersCustomResource?.Type === 'AWS::Lambda::Function';
+  const providerCreatedInCloud = hasHostedProviderResources(authCfnTemplate);
+
+  return lambdaCalloutCreatedInCloud && !providerCreatedInCloud;
+};
+
+const hasHostedProviderResources = (authCfnTemplate: Template | undefined): boolean => {
+  return (
+    authCfnTemplate?.Resources?.HostedUIFacebookProviderResource?.Type === 'AWS::Cognito::UserPoolIdentityProvider' ||
+    authCfnTemplate?.Resources?.HostedUIGoogleProviderResource?.Type === 'AWS::Cognito::UserPoolIdentityProvider' ||
+    authCfnTemplate?.Resources?.HostedUILoginWithAmazonProviderResource?.Type === 'AWS::Cognito::UserPoolIdentityProvider' ||
+    authCfnTemplate?.Resources?.HostedUISignInWithAppleProviderResource?.Type === 'AWS::Cognito::UserPoolIdentityProvider'
+  );
+};
+
+export const getHostedUIProviderCredsFromCloud = async (
+  resourceName: string,
+  providerMeta: $TSObject[],
+  updatedUIProviderCreds: $TSObject[],
+): Promise<$TSObject[]> => {
+  const userPoolId = getUserPoolId(resourceName);
+
+  const mapExistingProviderCreds = async (provider: $TSObject) => {
```

Comment:
Having a hard time figuring out the intent of this mapping. Maybe add a couple comments here for me and our future selves?